ykrt/src/mt.rs

```diff
@@ -286,12 +296,21 @@ impl MT {
             TransitionControlPoint::StopSideTracing(hl_arc, sti, parent) => {
                 // Assuming no bugs elsewhere, the `unwrap` cannot fail, because `StartTracing`
                 // will have put a `Some` in the `Rc`.
-                let thrdtrcr = THREAD_MTTHREAD.with(|mtt| mtt.thread_tracer.take().unwrap());
+                let (thrdtrcr, promotions) = THREAD_MTTHREAD.with(|mtt| {
+                    (
+                        mtt.thread_tracer.take().unwrap(),
+                        mtt.promotions.take().unwrap(),
```

Comment:
we should say why the unwraps are safe.

Reply:
The existing comment on line 296 explains.

Reply:
Oh right. I thought that was referring to something else because it's singular. Maybe wack and `s` on the end?

(Top-notch pedantry right here!)

Reply:
Fixed in ef0b3e11.